eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -5,7 +5,7 @@ parameters:
     type: string
   - name: SpecRepoCommit
     type: string
-    default: 'HEAD'
+    default: '$(Build.SourceBranch)'
```

Comment:
While I did suggest this, we should make sure we test this outside of the context of a PR to ensure it doesn't cause issues.

Reply:
This change has been verified. This is a [test run](https://dev.azure.com/azure-sdk/playground/_build/results?buildId=4503033&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e)